contrib/workers-bash-scripts/create-multiple-stream-writers.md

```diff
@@ -0,0 +1,143 @@
+# Creating multiple stream writers with a bash script
+
+This script creates multiple [stream writer](https://github.com/matrix-org/synapse/blob/develop/docs/workers.md#stream-writers) workers.
+
+Stream writers require both replication and HTTP listeners.
+
+It also prints out the example lines for Synapse main configuration file.
+
+Remember to route necessary endpoints directly to a worker associated with it.
+
+If you run the script as-is, it will create workers with the replication listener starting from port 8034 and another, "safe" listener starting from 8044. If you don't need all of stream writers listed in the script, just remove them from the ```STREAM_WRITERS``` array.
+
+```sh
+#!/bin/bash
+
+# Start with these replication and http ports.
+# The script loop starts with the exact port and then increments it by one.
+REP_START_PORT=8034
+HTTP_START_PORT=8044
+
+# Stream writer workers to generate. Feel free to add or remove them as you wish.
+# Event persister ("events") isn't included here as it does not require its
+# own HTTP listener.
+
+STREAM_WRITERS+=( "presence" "typing" "receipts" "to_device" "account_data" )
+
+NUM_WRITERS=$(expr ${#STREAM_WRITERS[@]})
+
+i=0
+
+while [ $i -lt "$NUM_WRITERS" ]
```

Comment:
isn't this more easily written

```
for stream in "${STREAM_WRITERS[@]}"
```

Reply:
I tried creating a **for** loop, similar to the one in the previous bash script. It seems that I have lots of trouble getting it to work (on bash 5.1.16). I either works incorrectly or I get errors (such as _operand expected_). On the other hand, **while** loop worked without any issues so I just went with it. I tried it the code above, too, with no luck.